src/modules/Dashboard/features/RespondentData/RespondentDataSummary/RespondentDataSummary.utils.tsx

```diff
@@ -586,13 +661,78 @@ export const compareActivityItem = (
             options: Object.values(sliderOptions),
           },
         },
-        answers: [...prevActivityItem.answers, ...answers],
+        answers: [...(prevActivityItem.answers as Answer[]), ...(answers as Answer[])],
+      };
+    }
+    case ItemResponseType.SingleSelectionPerRow:
```

Comment:
Please add unit tests to RespondentDataSummary.utils.test.tsx to cover this case.